web/modules/custom/asset_release/tests/src/Kernel/SyncFieldsFromParentSolutionTest.php

```diff
@@ -0,0 +1,247 @@
+      'field_is_state' => 'proposed',
+    ])->save();
+  }
+
+  /**
+   * Tests field synchronization between solution and release.
+   */
+  public function testSyncFieldsFromParentSolution() {
+    // Create a solution.
+    Rdf::create([
+      'rid' => 'solution',
+      'id' => 'http://example.com/solution',
+      'label' => 'Solution',
+      'field_is_state' => 'validated',
+      // Synchronized fields.
+      'field_is_description' => 'init description',
+      'field_is_solution_type' => 'http://example.com/eira/t1',
+      'field_is_contact_information' => 'http://example.com/contact1',
+      'field_is_owner' => 'http://example.com/owner1',
+      'field_is_related_solutions' => 'http://example.com/solution-related1',
+      'field_is_included_asset' => 'http://example.com/solution-related1',
+      'field_is_translation' => 'http://example.com/solution-related1',
+      'field_policy_domain' => 'http://example.com/policy-domain/d1',
+    ])->save();
+
+    // Create a release.
+    $release = Rdf::create([
+      'rid' => 'asset_release',
+      'id' => 'http://example.com/release',
+      'label' => 'release',
+      'field_isr_is_version_of' => 'http://example.com/solution',
+    ]);
+    $release->save();
+
+    // Check that the release fields were synchronized.
+    $this->assertEquals('init description', $release->field_isr_description->value);
+    $this->assertEquals('http://example.com/eira/t1', $release->field_isr_solution_type->target_id);
+    $this->assertEquals('http://example.com/contact1', $release->field_isr_contact_information->target_id);
+    $this->assertEquals('http://example.com/owner1', $release->field_isr_owner->target_id);
+    $this->assertEquals('http://example.com/solution-related1', $release->field_isr_related_solutions->target_id);
+    $this->assertEquals('http://example.com/solution-related1', $release->field_isr_included_asset->target_id);
+    $this->assertEquals('http://example.com/solution-related1', $release->field_isr_translation->target_id);
+    $this->assertEquals('http://example.com/policy-domain/d1', $release->field_policy_domain->target_id);
+
+    // Change the values of solution fields but reload the solution first, in
+    // order to get the last value of 'field_is_has_version' field.
+    Rdf::load('http://example.com/solution')
+      ->set('field_is_description', 'changed description')
+      ->set('field_is_solution_type', 'http://example.com/eira/t2')
+      ->set('field_is_contact_information', 'http://example.com/contact2')
+      ->set('field_is_owner', 'http://example.com/owner2')
+      ->set('field_is_related_solutions', 'http://example.com/solution-related2')
+      ->set('field_is_included_asset', 'http://example.com/solution-related2')
+      ->set('field_is_translation', 'http://example.com/solution-related2')
+      ->set('field_policy_domain', 'http://example.com/policy-domain/d2')
+      ->save();
+
+    // Reload the release.
```

Comment:
This is really nice! I can imagine we want to extend this further in the future to test updating the solution too...